src/arbitrary/_std/char.rs

```diff
@@ -0,0 +1,93 @@
+//-
+// Copyright 2017, 2018 The proptest developers
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Arbitrary implementations for `std::char`.
+
+use std::char::*;
+use std::iter::once;
+use std::ops::Range;
+
+use strategy::*;
+use strategy::statics::static_map;
+use arbitrary::*;
+
+macro_rules! impl_wrap_char {
+    ($type: ty, $mapper: expr) => {
+        arbitrary!($type, SMapped<char, Self>;
+            static_map(any::<char>(), $mapper));
+    };
+}
+
+impl_wrap_char!(EscapeDebug, char::escape_debug);
+impl_wrap_char!(EscapeDefault, char::escape_default);
+impl_wrap_char!(EscapeUnicode, char::escape_unicode);
+#[cfg(feature = "unstable")]
+impl_wrap_char!(ToLowercase, char::to_lowercase);
+#[cfg(feature = "unstable")]
+impl_wrap_char!(ToUppercase, char::to_uppercase);
+
+#[cfg(feature = "unstable")]
+arbitrary!(DecodeUtf8<<Vec<u8> as IntoIterator>::IntoIter>,
+    Flatten<Mapped<u16, SMapped<Vec<u8>, Self>>>;
+    {
+        use collection::size_range;
+        any::<u16>().prop_flat_map(|size| static_map(any_with::<Vec<u8>>(
```

Comment:
Why not just do `size_range(..65536)`? `prop_flat_map` is pretty hard on shrinking

Reply:
Fixed =)